engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -2947,12 +2947,25 @@ protected void createStoragePoolMappingsForVolumes(VirtualMachineProfile profile
             executeManagedStorageChecksWhenTargetStoragePoolNotProvided(targetHost, currentPool, volume);
             if (ScopeType.HOST.equals(currentPool.getScope()) || isStorageCrossClusterMigration(plan.getClusterId(), currentPool)) {
                 createVolumeToStoragePoolMappingIfPossible(profile, plan, volumeToPoolObjectMap, volume, currentPool);
-            } else {
+            } else if (shouldMapVolume(profile, volume, currentPool)){
                 volumeToPoolObjectMap.put(volume, currentPool);
             }
         }
     }
 
+    /**
+     * Returns true if it should map the volume for a storage pool to migrate.
+     * <br><br>
+     * Some context: VMware migration workflow requires all volumes to be mapped (even if volume stays on its current pool);
```

Comment:
The current condition checks for not KVM. Do we have to check for VMware only, to map all volumes?

Reply:
@sureshanaparti thanks again for reviewing.

My main idea on isolating this on KVM is due to the fact that I was able to reproduce/debug this issue only on KVM. Additionally, I am taking into account the `KvmNonManagedStorageDataMotionStrategy` migration process.

Therefore, considering the review and tests from @shwstppr with the fact that I am seeing this in KVM I changed the mapping just for the KVM.
The question would be, does this issue is also happening on XenServer? If anyone can check it on a XenServer env it would be great.